velox/exec/LocalPlanner.cpp

```diff
@@ -292,6 +292,52 @@ void LocalPlanner::plan(
         }
       }
     }
+
+    // Detect joins running in grouped execution mode having sources running in
+    // ungrouped execution mode. In such case the join bridge would be between
+    // grouped execution and ungrouped execution and needs special care to be
+    // detected by both pipelines.
+    for (auto& factory : *driverFactories) {
```

Comment:
nit: this method is rather long; consider extracting this new logic into a helper function